system/API/ResponseTrait.php

```diff
@@ -386,32 +387,27 @@ protected function format($data = null)
 			return $data;
 		}
 
-		$config = new Format();
-		$format = "application/$this->format";
+		$mime = "application/$this->format";
 
 		// Determine correct response type through content negotiation if not explicitly declared
 		if (empty($this->format) || ! in_array($this->format, ['json', 'xml']))
 		{
-			$format = $this->request->negotiate('media', $config->supportedResponseFormats, false);
+			$config = new Format();
+			$mime = $this->request->negotiate('media', $config->supportedResponseFormats);
 		}
 
-		$this->response->setContentType($format);
+		$this->response->setContentType($mime);
 
-		// if we don't have a formatter, make one
-		if (! isset($this->formatter))
-		{
-			// if no formatter, use the default
-			$this->formatter = $config->getFormatter($format);
-		}
```

Comment:
We have to check if there is `$this->formatter` already available. And only if there isn't, we should load it via Service.
```php
$this->formatter = Services::format()->getFormatter($mime);
```